test/support.py

```diff
@@ -272,6 +287,45 @@ def load_cpython_module(self, name, so_filename):
         return module
 
 
+class PythonSubprocessRunner:
+    def __init__(self, verbose, hpy_abi):
+        self.verbose = verbose
+        self.hpy_abi = hpy_abi
+
+    def run(self, mod, code):
+        """ Starts new subprocess that loads given module as 'mod' using the
+            correct ABI mode and then executes given code snippet. Use
+            "--subprocess-v" to enable logging from this.
+        """
+        import subprocess
+        env = os.environ.copy()
+        pythonpath = [os.path.dirname(mod.so_filename)]
+        if 'PYTHONPATH' in env:
+            pythonpath.append(env['PYTHONPATH'])
+        env["PYTHONPATH"] = os.pathsep.join(pythonpath)
+        if self.hpy_abi in ['universal', 'debug']:
+            # HPy module
+            load_module = "import sys;" + \
+                          "import hpy.universal;" + \
+                          "mod = hpy.universal.load('{name}', '{so_filename}', debug={debug});"
+            escaped_filename = mod.so_filename.replace("\\", "\\\\")  # Needed for Windows paths
+            load_module = load_module.format(name=mod.name, so_filename=escaped_filename,
+                                             debug=self.hpy_abi == 'debug')
+        else:
+            # CPython module
+            assert self.hpy_abi == 'cpython'
+            load_module = "import {} as mod;".format(mod.name)
+        if self.verbose:
+            print("\n---\nExecuting in subprocess: {}\n".format(load_module + code))
+        result = subprocess.run([sys.executable, "-c", load_module + code], env=env,
+                                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        if self.verbose:
+            out = result.stdout.decode('ascii')
+            err = result.stderr.decode('ascii')
+            print("\n---\n{out}\n{err}\n---\n".format(out=out, err=err))
+        return result
```

Comment:
this looks potentially problematic: if for any reason the subprocess prints a non-ascii char, this would throw `UnicodeDecodeError`, I think.
Probably it's safer to `.decode('latin-1')` (which always works because it maps the whole range 0-255, even if it might result in funny question marks).

Reply:
I sneaked the fix of this into: https://github.com/hpyproject/hpy/pull/259